amalgamate.py

```diff
@@ -138,112 +136,158 @@ def create_version_header(version, file_name="version.h"):
-        help="Output *.cpp source path.",
+        help="Output C++ source file name.",
     )
     parser.add_argument(
-        "-i",
-        dest="header_path",
-        action="store",
-        default="pyne.h",
-        help="Output header path.",
+        "-i", dest="header_name", default="pyne.h", help="Output header file name."
     )
     parser.add_argument(
         "-f",
         dest="files",
         nargs="+",
-        help="Files to amalgamate.",
         default=DEFAULT_FILES,
+        help="Input files to amalgamate.",
+    )
+    parser.add_argument(
+        "-o",
+        dest="output_dir",
+        default=".",
+        help="Output directory for generated files.",
     )
-    ns = parser.parse_args()
+
+    args = parser.parse_args()
+    output_dir = os.path.abspath(args.output_dir)
+
     version = get_version()
     create_version_header(version)
-    # header file
-    hdr = AmalgamatedFile(ns.header_path)
-    hdr.append_line("// PyNE amalgated header http://pyne.io/")
-    hdr.append_line("#ifndef PYNE_52BMSKGZ3FHG3NQI566D4I2ZLY")
-    hdr.append_line("#define PYNE_52BMSKGZ3FHG3NQI566D4I2ZLY")
-    hdr.append_line("")
-    hdr.append_line("#define PYNE_IS_AMALGAMATED")
-    hdr.append_line("")
-    for f in ns.files:
-        _, ext = os.path.splitext(f)
-        if ext in SOURCE_EXTS:
-            continue
-        hdr.append_file(f)
-    hdr.append_line("#endif  // PYNE_52BMSKGZ3FHG3NQI566D4I2ZLY")
-
-    # source file
-    src = AmalgamatedFile(ns.source_path)
-    src.append_line("// PyNE amalgated source http://pyne.io/")
-    src.append_line(
-        '#include "{0}"'.format(
-            os.path.relpath(ns.header_path, os.path.dirname(ns.source_path))
-        )
-    )
-    src.append_line("")
-    for f in ns.files:
-        _, ext = os.path.splitext(f)
+
+    header_path = os.path.join(output_dir, args.header_name)
+    source_path = os.path.join(output_dir, args.source_name)
+
+    # First, figure out which files are headers
+    headers_to_amalgamate = {
+        f for f in args.files if os.path.splitext(f)[1] in HEADER_EXTS
+    }
+
+    # Header and Source Generation
+    header = AmalgamatedFile(header_path)  # Header doesn't need to filter itself
+    header.append_line("// Amalgamated PyNE header - http://pyne.io/")
+    header.append_line("#ifndef PYNE_AMALGAMATED_HEADER")
+    header.append_line("#define PYNE_AMALGAMATED_HEADER\n")
+    header.append_line("#define PYNE_IS_AMALGAMATED\n")
+
+    # Pass the list of amalgamated headers to the source file constructor
+    source = AmalgamatedFile(source_path, amalgamated_headers=headers_to_amalgamate)
+    source.append_line("// Amalgamated PyNE source - http://pyne.io/")
+    rel_header_path = os.path.relpath(header_path, start=os.path.dirname(source_path))
+
+    # Ensure forward slashes for cross-platform compatibility in #include
+    rel_header_path = rel_header_path.replace("\\", "/")
+    source.append_line(f'#include "{rel_header_path}"\n')
```

Comment:
I feel like it would be good to have one more blank line here
```suggestion
    source.append_line(f'#include "{rel_header_path}"\n\n')
```